internal/controller/pkg/resolver/reconciler_test.go

```diff
@@ -489,6 +490,52 @@ func TestReconcile(t *testing.T) {
 				r: reconcile.Result{Requeue: false},
 			},
 		},
+		"SuccessfulCreateMissingDependencyDigest": {
```

Comment:
nit:

We may want to add `SuccessfulNoMissing` and `ErrorCreateMissingDependency` variants of the same to get it a bit more validated.

Reply:
added